core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -114,6 +123,19 @@ class AdminClient(val time: Time,
     listAllGroupsFlattened.filter(_.protocolType == ConsumerProtocol.PROTOCOL_TYPE)
   }
 
+  def listAllBrokerVersionInfo(): Map[Node, Try[NodeApiVersions]] = {
+    findAllBrokers.map {
+      case broker =>
+        broker -> {
+          try {
+            Success(new NodeApiVersions(getApiVersions(broker)))
+          } catch {
+            case e: Exception => Failure(e)
+          }
```

Comment:
You could replace the `try/catch` with `Try(new NodeApiVersions(getApiVersions(broker)))`.